src/connect/command.cpp

```diff
@@ -133,19 +145,15 @@ Command Command::parse_json_command(CommandId id, const string_view &body, Share
             data = BrokenCommand {};
         }
     } else if (auto *info = get_if<SendFileInfo>(&data); info != nullptr) {
-        if (has_path) {
-            info->path = SharedPath(move(buff));
-        } else {
-            // Missing parameters
-            data = BrokenCommand {};
-        }
+        get_path(info->path);
     } else if (auto *start = get_if<StartPrint>(&data); start != nullptr) {
-        if (has_path) {
-            start->path = SharedPath(move(buff));
-        } else {
-            // Missing parameters
-            data = BrokenCommand {};
-        }
+        get_path(start->path);
+    } else if (auto *del_file = get_if<DeleteFile>(&data); del_file != nullptr) {
+        get_path(del_file->path);
+    } else if (auto *del_folder = get_if<DeleteFolder>(&data); del_folder != nullptr) {
+        get_path(del_folder->path);
+    } else if (auto *create_folder = get_if<CreateFolder>(&data); create_folder != nullptr) {
+        get_path(create_folder->path);
```

Comment:
:+1: for the deduplication.

In theory, we could have something like a „cycle“ over a typelist… but I don't think such abominations really belong to a real code.